extensions/markdown/README.md

```diff
@@ -0,0 +1,30 @@
+## Using Markdown-it Plugins
+
+To use a `markdown-it` plugin, goto the directory containing the `markdown-it` extension.  This will
+be found in the `extensions` folder in your vscode installation.  Install the `markdown-it` plugin
+as a node module.  For example, to install the `KaTeX` markdown-it extension you would simply type:
+
+``` sh
+npm install markdown-it-katex
+```
+
+Next, add the plugin into your [settings configuration](https://code.visualstudio.com/docs/customization/userandworkspace)
+under the `"markdown.plugs"` namespace.  This setting should contain a list for each `markdown-it` extension you wish
+to install.  The list should contain an object describing the name of the module, a list of styles that your extentsion
+may require, and options that you wish to pass to your extension.  For example, while installing the `markdown-it-katex`
+extension we would include this in your settings configuration:
+
+``` json
+    "markdown.plugins": [
+        {
+            "name": "markdown-it-katex",
+            "styles": [
```

Comment:
Oh, you are right, I made a mistake.  The plugins do not overwrite previously loaded plugins even if they are already loaded, as an example:

```
var mdnh = require('markdown-it-named-headers');
var md = require('markdown-it')().use('mdnh').use('mdnh');

md.render('# Test')
```
will return '<h1 id="test" id="test">Test</h1>'.  I will go ahead an do what you suggested below, make a `builtInPlugins` list and merge it with `userPlugins` as you have suggested and then clean up the README.md to clarify the order in which events occur and that the order may override other settings.  Thanks for looking at this!